### ISSUE TITLE
Script building the release package

### ISSUE THREAD

**Issue by douglowe** [LightForm-group/KWN-deformation#15]
Create Cmake script for building the package. Ensure this works for OSX / linux (using make), and Windows (using either ninja or nmake).

**Comment by douglowe** [comment]
Reorganisation in preparation for this has been made in this PR: [https://github.com/LightForm-group/KWN-deformation/pull/23](https://github.com/LightForm-group/KWN-deformation/pull/23)

**Comment by douglowe** [comment]
I've renamed this issue, as I'm not 100% certain CMake is the right choice for the project. An alternative choice would be the new [fortran package manager](https://fpm.fortran-lang.org/en/index.html), which seems much lighter weight than CMake.

I've created two branches, to show the differences (though with a single source file, the complexity of both is low):
CMake: https://github.com/LightForm-group/KWN-deformation/tree/cmake_initialisation
fpm: https://github.com/LightForm-group/KWN-deformation/tree/fpm_initialisation

Biggest difference is that fpm uses a predefined directory structure to locate the main program code (in `app`), and the modules required by the program code (in `src`). This means that the fpm configuration file doesn't need updating if the modules are changed, it should pick this all up automatically. An example of this can be seen at the [start of my module work](https://github.com/LightForm-group/KWN-deformation/tree/9c937f4cec39da1bcd431cf637610be8568b1075), where I've moved the precision and parameter code to separate files in `app`, and the model still compiles without requiring any changes to the `fpm.toml` or `install.sh` files.

For now I'll continue development using fpm - if we decide to continue with CMake it won't require much more work to switch to this.
